README.md

```diff
@@ -1,2 +1,104 @@
 # re-route
-Routing wrapper library for SPAs based on re-frame.
+
+Routing wrapper library for SPAs based on [re-frame](https://github.com/day8/re-frame) and [reitit](https://github.com/metosin/reitit).
+
+In addition to standard navigation functionality, this library comes with built-in functionality from navigation prevention, e.g. to warn the user of unsaved input changes.
+
+## Installation
+
+To use re-route as a dependency in your ClojureScript project, add the following to your `deps.edn` file:
+
+```clojure
+{ com.yetanalytics/re-route {:mvn/version "0.0.1"} }
+```
+
+**NOTE:** re-route assumes that your browser supports the [History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API).
+
+## API
+
+All API functions are located in the `com.yetanalytics.re-route` namespace, which we will alias to `re-route`.
+
+### Effect Handlers
+
+#### `[::re-route/init routes default-route prevent-nav-opts]`
+
+Starts a reitit router and initializes re-route's values in the re-frame app-db.
+
+`routes` is a reitit-compatible vector of route maps; a standard route map looks something like:
+```clojure
+{:name        :route-name
+ :view        (fn [args] ...)
+ :parameters  { ... }
+ :controllers [{:start (fn [params] ...)
+                :stop  (fn [params] ...)}]}
+```
+where `:controllers` contains the `:start` and `:stop` functions that are called when entering and leaving the route, respectively. `:view` is a property unique to re-route that specifies a Reagent render function.
```

Comment:
what's `:identity`

Reply:
That is a reitit keyword, to define functions that return custom parameters to your `:start` and `:stop` functions (if you need something more complex than `:parameters`).